app/src/main/java/pl/ludek/smat/home/pool_office_client/presentation/MainActivityViewModel.kt

```diff
@@ -1,81 +1,36 @@
 package pl.ludek.smat.home.pool_office_client.presentation
 
-import android.util.Log
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import pl.ludek.smat.home.pool_office_client.data.apiservice.PoolInfoClient
-import pl.ludek.smat.home.pool_office_client.data.apiservice.InitializationStateRelay
-import pl.ludek.smat.home.pool_office_client.data.apiservice.PoolInfoData
-import pl.ludek.smat.home.pool_office_client.data.apiservice.RelayData
-import retrofit2.Call
-import retrofit2.Callback
-import retrofit2.Response
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import pl.ludek.smat.home.pool_office_client.data.apiservice.*
 
 private val TAG = MainActivityViewModel::class.java.simpleName
 const val RELAY_ID_ALL = 256
 
 class MainActivityViewModel : ViewModel() {
 
     private val poolInfoClient = PoolInfoClient
-    val singleRelayStateData = MutableLiveData<RelayData>()
-    val completeRelayStateData = MutableLiveData<Array<Boolean>>()
-
-    val poolInfoData = MutableLiveData<PoolInfoData>()
+    val singleRelayStateData = MutableLiveData<NetworkResult<RelayData>>()
+    val completeRelayStateData = MutableLiveData<NetworkResult<InitializationStateRelay>>()
+    val poolInfoData =MutableLiveData<NetworkResult<PoolInfoData>>()
 
     fun updatePoolInfo() {
-        poolInfoClient.getSensorData().enqueue(object : Callback<PoolInfoData> {
-            override fun onResponse(call: Call<PoolInfoData>, response: Response<PoolInfoData>) {
-                if (response.isSuccessful) {
-                    poolInfoData.value = response.body()
-                } else {
-                    poolInfoData.value = PoolInfoData(0.0f,0.0f,0.0f,0.0f,true)
-                }
-            }
-
-            override fun onFailure(call: Call<PoolInfoData>, t: Throwable) {
-                poolInfoData.value = PoolInfoData(0.0f,0.0f,0.0f,0.0f,true)
-                Log.d(TAG, t.message.toString())
-            }
-        })
+        GlobalScope.launch {
```

Comment:
The coroutines scopes are rather an extended topic, and you will become familiar with them in the future but for now please just remember to always use `viewModelScope` in `ViewModel`s and be careful with using the `GlobalScope`.
Please replace all the `GlobalScope.launch` with the `viewModelScope.launch`. This will cause the network calls to be canceled if the `ViewModel` is being stopped (for instance when you close the app).